commands/core/queue.drush.inc

```diff
@@ -116,8 +116,11 @@ function drush_queue_run($queue_name) {
   $end = time() + (isset($info['cron']['time']) ? $info['cron']['time'] : 15);
   $queue = drush_queue_get($queue_name);
   while (time() < $end && ($item = $queue->claimItem())) {
-    $function($item->data);
+    $result = $function($item->data);
     $queue->deleteItem($item);
```

Comment:
Worker callback return values are a new concept so I think we get to define what's right. Right now throwing an exception is the only way to keep the item from being deleted, which seems appropriate.